### PR TITLE
added tests 

### PR DESCRIPTION
**Pull request by metelska-anastasiia**
for ShoppingCart, CartItem, User, Order, OrderItem entities (repository, service, controller)
Снимок экрана 2023-11-08 в 11 43 33

### CODE REVIEW

File: src/test/java/mate/academy/controller/AuthenticationControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.http.MediaType;
+import org.springframework.jdbc.datasource.init.ScriptUtils;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class AuthenticationControllerTest {
```
Comment (Ivan95kos) [inline src/test/java/mate/academy/controller/AuthenticationControllerTest.java]:
```suggestion
class AuthenticationControllerIntegrationTest {
```

File: src/test/java/mate/academy/controller/AuthenticationControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    }
+}
+
+
+
+
+
+
+
+
```
Comment (Ivan95kos) [inline src/test/java/mate/academy/controller/AuthenticationControllerTest.java]: lines
```suggestion
```

File: src/test/java/mate/academy/controller/OrderControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class OrderControllerTest {
```
Comment (Ivan95kos) [inline src/test/java/mate/academy/controller/OrderControllerTest.java]:
```suggestion
class OrderControllerIntegrationTest {
```

File: src/test/java/mate/academy/controller/ShoppingCartControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class ShoppingCartControllerTest {
```
Comment (Ivan95kos) [inline src/test/java/mate/academy/controller/ShoppingCartControllerTest.java]: the same

File: src/test/java/mate/academy/repository/BookRepositoryTest.java [hunk 5]
```diff
@@ -2,30 +2,43 @@
...
 import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.PageRequest;
 import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
 import org.springframework.test.context.jdbc.Sql;
 
 @DataJpaTest
 @AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
 class BookRepositoryTest {
```
Comment (Ivan95kos) [inline src/test/java/mate/academy/repository/BookRepositoryTest.java]: also
```suggestion
class BookRepositoryIntegrationTest {
```

File: src/test/java/mate/academy/repository/CartItemRepositoryTest.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.jdbc.Sql;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class CartItemRepositoryTest {
```
Comment (Ivan95kos) [inline src/test/java/mate/academy/repository/CartItemRepositoryTest.java]: also

File: src/test/java/mate/academy/repository/OrderRepositoryTest.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.test.context.jdbc.Sql;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class OrderRepositoryTest {
```
Comment (Ivan95kos) [inline src/test/java/mate/academy/repository/OrderRepositoryTest.java]: +

File: src/test/java/mate/academy/repository/ShoppingCartRepositoryTest.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+import mate.academy.repository.cart.ShoppingCartRepository;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.jdbc.Sql;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class ShoppingCartRepositoryTest {
```
Comment (Ivan95kos) [inline src/test/java/mate/academy/repository/ShoppingCartRepositoryTest.java]: +

File: src/test/java/mate/academy/repository/UserRepositoryTest.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.jdbc.Sql;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class UserRepositoryTest {
```
Comment (Ivan95kos) [inline src/test/java/mate/academy/repository/UserRepositoryTest.java]: also

File: src/test/java/mate/academy/repository/BookRepositoryTest.java [hunk 10]
```diff
@@ -65,4 +78,82 @@ void findAll_WithPagination_ShouldReturnPage() {
...
+                        .setAuthor("Author 1")
+                        .setIsbn("ISBN-123456")
+                        .setPrice(BigDecimal.valueOf(100))
+                        .setDescription("Description for Book 1")
+                        .setCoverImage("image1.jpg")
+                        .setCategories(Set.of(new Category()
+                                .setId(1L)
+                                .setName("Poetry")
+                                .setDescription("Poems that you will love"))
+                        )
```
Comment (Ivan95kos) [inline src/test/java/mate/academy/repository/BookRepositoryTest.java]: it is better to move the common data to the private methods,
fix it everywhere

File: src/test/java/mate/academy/service/OrderServiceImplTest.java [hunk 11]
```diff
@@ -0,0 +1,258 @@
...
+                .setOrderItems(Set.of(orderItem))
+                .setId(1L);
+        cartItem = new CartItem()
+                .setBook(book)
+                .setQuantity(1)
+                .setId(1L);
+        orderItemResponseDto = new OrderItemResponseDto()
+                .setId(1L)
+                .setQuantity(1)
+                .setBookId(1L);
```
Comment (Ivan95kos) [inline src/test/java/mate/academy/service/OrderServiceImplTest.java]: Such code is better to read, 
and you can use these methods wherever you need
```suggestion
        user = prepareUser();
        book = prepareBook();
        orderItem = prepareOrderItem();
        order = prepareOrder();
        cartItem = prepareCartItem();
        orderItemResponseDto = prepareOrderItemResponseDto();
```

File: src/test/java/mate/academy/service/OrderServiceImplTest.java [hunk 12]
```diff
@@ -0,0 +1,258 @@
...
+    @InjectMocks
+    private OrderServiceImpl orderService;
+    @Mock
+    private Authentication authentication;
+    private User user;
+    private Book book;
+    private OrderItem orderItem;
+    private Order order;
+    private CartItem cartItem;
+    private OrderItemResponseDto orderItemResponseDto;
```
Comment (Ivan95kos) [inline src/test/java/mate/academy/service/OrderServiceImplTest.java]: It is better not to use the general test data
```suggestion
```

File: src/test/java/mate/academy/service/OrderServiceImplTest.java [hunk 13]
```diff
@@ -0,0 +1,258 @@
...
+    @Mock
+    private OrderItemRepository orderItemRepository;
+    @Mock
+    private OrderMapper orderMapper;
+    @Mock
+    private OrderItemMapper orderItemMapper;
+    @InjectMocks
+    private OrderServiceImpl orderService;
+    @Mock
+    private Authentication authentication;
```
Comment (Ivan95kos) [inline src/test/java/mate/academy/service/OrderServiceImplTest.java]: change the order
```suggestion
    @Mock
    private Authentication authentication;
    @InjectMocks
    private OrderServiceImpl orderService;
```

**Review by Ivan95kos — CHANGES_REQUESTED** [review]
left a few comments

File: src/test/java/mate/academy/controller/AuthenticationControllerIntegrationTest.java [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+    @Autowired
+    private AuthenticationService authenticationService;
+    @Autowired
+    private JwtUtil jwtUtil;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired DataSource dataSource,
+            @Autowired WebApplicationContext applicationContext
+    ) {
```
Comment (Sanderfuck) [inline src/test/java/mate/academy/controller/AuthenticationControllerIntegrationTest.java]: Why you Autowire here?

File: src/test/java/mate/academy/controller/AuthenticationControllerIntegrationTest.java [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+
+    @BeforeEach
+    void setUp(@Autowired DataSource dataSource) {
+        setupDatabase(dataSource);
+    }
+
+    @AfterEach
+    void afterEach(@Autowired DataSource dataSource) {
+        teardown(dataSource);
+    }
```
Comment (Sanderfuck) [inline src/test/java/mate/academy/controller/AuthenticationControllerIntegrationTest.java]: I think exist better approach for init and clean database

File: src/test/java/mate/academy/controller/AuthenticationControllerIntegrationTest.java [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+                .setRepeatPassword("Newuser12345")
+                .setFirstName("User")
+                .setLastName("New");
+
+        UserResponseDto expected = new UserResponseDto()
+                .setId(3L)
+                .setEmail(request.getEmail())
+                .setFirstName(request.getFirstName())
+                .setLastName(request.getLastName());
+
```
Comment (Sanderfuck) [inline src/test/java/mate/academy/controller/AuthenticationControllerIntegrationTest.java]: You can create template request, users, etc in some common file and use it every test what you need for increase readability

File: src/test/java/mate/academy/controller/BookControllerIntegrationTest.java [hunk 17]
```diff
@@ -308,4 +284,16 @@ void updateBook_WithValidIdAndCreateBookRequestDto_Success() throws Exception {
...
+        return new BookDto()
+                .setId(1L)
+                .setTitle("Book 1")
+                .setAuthor("Author 1")
+                .setIsbn("ISBN-123456")
+                .setPrice(BigDecimal.valueOf(100))
+                .setDescription("Description for Book 1")
+                .setCoverImage("image1.jpg")
+                .setCategoryIds(Set.of());
+    }
```
Comment (Sanderfuck) [inline src/test/java/mate/academy/controller/BookControllerIntegrationTest.java]: Replace to common class with templates

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

**Review by kozhukhovsky — APPROVED** [review]

**Pull request merged**